### ISSUE TITLE
[Integration] RustPython slow to start and exit with error

### ISSUE THREAD

**Issue by KernelErr** [WasmEdge/WasmEdge#996]
Hi, I compiled RustPython for `wasm32-wasi` target. However, I found that WasmEdge would take much more time to have a Python shell than wasmtime and wasmer.

WasmEdge version: 0.9.0

Time:

- WasmEdge: ~40s
- Wasmtime: ~1s
- Wasmer: ~5s

Module File(PR: https://github.com/RustPython/RustPython/pull/3514):

[rustpython.wasm.zip](https://github.com/WasmEdge/WasmEdge/files/7826880/rustpython.wasm.zip)

**Comment by KernelErr** [comment]
Update: Using `wasmedgec` to compile wasm to so file can have the same starting time. I would point it out in later document.

**Comment by KernelErr** [comment]
After `exit()`, wasmedge would throw out an unexpected error. The platform is Linux and wasmedge version is 0.9.0.

![image](https://user-images.githubusercontent.com/45716019/148503019-8830b0cb-041a-4eb4-a7cc-8e9675bf34b1.png)

**Comment by juntao** [comment]
Let's do a profiler report like this to identify the bottlenecks. 

https://github.com/WasmEdge/WasmEdge/issues/473

**Comment by KernelErr** [comment]
I compiled the latest commit(eda17462c3bafc432a1f70e5886a99ba7dae0912) and ran valgrind to check bottleneck:

```
valgrind --tool=callgrind ./tools/wasmedge/wasmedge ./rustpython.wasm -c "exit(0)"
```

There are some points I found:

- The latest code would take a much much longer time to start. (After several seconds, I still couldn't get a Python shell)
- Compile to `so` file, we can get Python shell in seconds.
- After executing `exit(0)` in Python, there is still an error. `_start` functions should the entrance of Rust programs.

![image](https://user-images.githubusercontent.com/45716019/148521770-e8eee92e-6773-484f-a1ef-c3ac07509756.png)

![callgrind](https://user-images.githubusercontent.com/45716019/148521156-d8993346-71ef-4c56-8304-2fd364e43440.png)
